src/tritonserver.cc

```diff
@@ -2962,8 +2960,16 @@ TRITONSERVER_Error*
 TRITONSERVER_MetricDelete(TRITONSERVER_Metric* metric)
 {
 #ifdef TRITON_ENABLE_METRICS
-  delete reinterpret_cast<tc::Metric*>(metric);
-  return nullptr;  // Success
+  auto lmetric = reinterpret_cast<tc::Metric*>(metric);
+  if (lmetric->Family() == nullptr) {
+    return TRITONSERVER_ErrorNew(
+        TRITONSERVER_ERROR_INTERNAL,
+        "MetricFamily reference was invalidated before Metric was deleted. "
```

Comment:
I think we should update the documentation for the metrics API to mention that it is best to delete all the metrics corresponding to a metric family before deleting the metric family. Otherwise, an error object will be received.

Reply:
Great suggestion, done